magma/coreirModuleWrapper.py

```diff
@@ -0,0 +1,17 @@
+from .backend.coreir_ import CoreIRBackend
+from .circuit import Circuit
+from magma import cache_definition
+
+@cache_definition
+def DefineModuleWrapper(cirb: CoreIRBackend, coreirModule):
+    class ModuleWrapper(Circuit):
+        name = coreirModule.name
+        IO = cirb.get_ports_as_list(cirb.get_ports(coreirModule.type))
+        wrappedModule = coreirModule
+    return ModuleWrapper
+
+def ModuleFromGeneratorWrapper(cirb: CoreIRBackend, namespace: str, generator: str,
+                               dependentNamespaces: list, genargs = {}, modargs = {}):
+    moduleToWrap = cirb.context.import_generator(namespace,generator)(**genargs)
+    cirb.context.run_passes(["rungenerators"], [namespace] + dependentNamespaces)
+    return DefineModuleWrapper(cirb, moduleToWrap)(**modargs)
```

Comment:
Minor nitpick, and this is probably annoying for your existing code, but can we change this module name to be `module_wrapper` (consistent with general Python style: https://www.python.org/dev/peps/pep-0008/#package-and-module-names) and can we maybe move it to a new submodule called `frontend`?

So using it would look something like
```
from magma.frontend.coreir import DefineModuleWrapper
```